virttest/utils_misc.py

```diff
@@ -2215,30 +2215,38 @@ def get_qemu_binary(params):
     """
     Get the path to the qemu binary currently in use.
     """
+    # _cmd => full command (including VAR=value, ...)
+    # _bin => binary path
+    params_qemu_cmd = params.get("qemu_binary", "qemu")
+    params_qemu_bin = process.binary_from_shell_cmd(params_qemu_cmd)
+
     # Update LD_LIBRARY_PATH for built libraries (libspice-server)
-    qemu_binary_path = get_path(_get_backend_dir(params),
-                                params.get("qemu_binary", "qemu"))
+    abs_qemu_bin = get_path(_get_backend_dir(params), params_qemu_bin)
 
-    if not os.path.isfile(qemu_binary_path):
-        logging.debug('Could not find params qemu in %s, searching the '
-                      'host PATH for one to use', qemu_binary_path)
+    if not os.path.isfile(abs_qemu_bin):
+        logging.warning('Could not find params qemu in %s, searching the '
+                        'host PATH for one to use', abs_qemu_bin)
         try:
-            qemu_binary = utils_path.find_command('qemu-kvm')
-            logging.debug('Found %s', qemu_binary)
+            qemu_cmd = utils_path.find_command('qemu-kvm')
+            logging.debug('Found %s', qemu_cmd)
         except utils_path.CmdNotFoundError:
-            qemu_binary = utils_path.find_command('kvm')
-            logging.debug('Found %s', qemu_binary)
+            qemu_cmd = utils_path.find_command('kvm')
+            logging.debug('Found %s', qemu_cmd)
     else:
         library_path = os.path.join(
             _get_backend_dir(params), 'install_root', 'lib')
+        if abs_qemu_bin != params_qemu_bin:
+            abs_qemu_cmd = params_qemu_cmd.replace(params_qemu_bin,
+                                                   abs_qemu_bin)
+        else:
+            abs_qemu_cmd = params_qemu_cmd
         if os.path.isdir(library_path):
             library_path = os.path.abspath(library_path)
-            qemu_binary = ("LD_LIBRARY_PATH=%s %s" %
-                           (library_path, qemu_binary_path))
+            qemu_cmd = ("LD_LIBRARY_PATH=%s %s" % (library_path, abs_qemu_cmd))
```

Comment:
One thought here: shouldn't we try to implement a proper way to set a given environment before executing important binaries such as qemu? Or even a generic way to any binary. I feel that the handling of `LD_LIBRARY_PATH` is really out of place.  That also means that the further handling of environment variables in the command line would not need `shell=True`, just a proper passing of the environment.

Still, I know this is the kind of situation where a big enough change can open a can of worms... Thoughts?